### PR TITLE
Pattern matching tuples

### PR DESCRIPTION
**Pull request by mutablestate**
I simply ran through the doc tests in sequence pattern matching.

### CODE REVIEW

File: rock_paper_scissors/lib/rock_paper_scissors.ex [hunk 1]
```diff
@@ -18,6 +18,10 @@ defmodule RockPaperScissors do
       iex> RockPaperScissors.score({:p1, :scissors}, {:p2, :rock})
       :p2
   """
-  def score(t1, t2) do
-  end
+  @spec score(tuple, tuple) :: atom
+  def score(t1, t2), do: _score(t1, t2)
+  defp _score({_, w1}, {_, w2}) when w1 == w2, do: :draw
```
Comment (citizen428) [inline rock_paper_scissors/lib/rock_paper_scissors.ex:23]: You can simplify this one further.

**Comment by citizen428** [comment]
You are not covering several scenarios, e.g.

``` elixir
RockPaperScissors.score({:p1, :paper}, {:p2, :scissors})
** (FunctionClauseError) no function clause matching in RockPaperScissors._score/2
    iex:19: RockPaperScissors._score({:p1, :paper}, {:p2, :scissors})
```

**Comment by mutablestate** [comment]
Thanks for picking up on the `:draw` match. I didn't know I could pattern match the same argument by naming it the same. I will remove the conditional.

I was hoping someone would pick me up on the lack of coverage for all scenarios as I wanted to bring up a discussion point ;-) 

Using a TDD style to pass the tests my solution is incomplete, however, the doctests don't cover other scenarios / edge cases. From my understanding doctests are normally used to give a quick usage example to 'document' a function and a quick sanity check, whereas unit tests should be used for all edge cases. Any thoughts?

**Comment by citizen428** [comment]
I think using the same variable to force the 2 values to be equal is much nicer than the guard.

As for the tests, I agree. Enumerating all options in the doctest seems like too much. I think I'd actually use some quote/unquote magic to generate all test cases in a proper unit test.

**Comment by riverrun** [comment]
I was going to mention the guard. Using the same variable to check for equality is another great aspect of pattern matching.
As for testing, in this case it would be quite easy to write all the tests as there are only six winning cases, but I agree that I would normally put them in the unit tests, rather than doctests.
